### ISSUE TITLE
"TypeError: 'range' object does not support item assignment"

### ISSUE THREAD

**Issue by Jessi5177** [rlopt/l2i#6]
I made an error in line 1167 of ml_opt.py when I surfaced this paper， the error message showed  "TypeError: 'range' object does not support item assignment". I would like to ask  this question is because of my python's version or other reasons.

**Comment by Jessi5177** [comment]
I found that the problem was that the range function in the code had to be converted to a list,for example, range(0,n) must write as list(range(0,n)).But I don't know why.

**Comment by YangXiwang** [comment]
if config.use_attention_embedding and num_states > config.batch_size:
config.batch_size = 1000
May I ask why using less than 1000 states？

**Comment by xxyun** [comment]
I also have this issue. Have you already solved it?

**Comment by YangXiwang** [comment]
No. I just learn their method but doesn't solve it. I find some code can't be 100% realized. It is a pity.
Are you also interested in RL for CO？



---Original---
From: ***@***.***&gt;
Date: Sun, Nov 5, 2023 13:13 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [rlopt/l2i] "TypeError: 'range' object does not support itemassignment" (#6)




 
I also have this issue. Have you already solved it?
